modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/ComponentLocationVisitor.java

```diff
@@ -209,7 +209,8 @@ private boolean isMunitFlowIdentifier(ComponentModel componentModel) {
   private boolean isRootProcessorScope(ComponentModel componentModel) {
     ComponentIdentifier identifier = componentModel.getIdentifier();
     return identifier.equals(FLOW_IDENTIFIER) || identifier.equals(MUNIT_BEFORE_SUITE_IDENTIFIER)
-        || identifier.equals(SUBFLOW_IDENTIFIER) ||
+        || identifier.equals(SUBFLOW_IDENTIFIER) || identifier.equals(CHOICE_WHEN_COMPONENT_IDENTIFIER)
```

Comment:
create an issue and add a TODO to review this situation with other components (try and scatter-gather pop into my mind right now.)